catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -409,91 +489,126 @@ private void assertReplies(List<Mutable> replies) {
                     .get("format"), is("DATE"));
-            assertThat((String) typeInfo.get("modified")
+            assertThat(typeInfo.get("modified")
                     .get("format"), is("DATE"));
-            assertThat((String) typeInfo.get("created")
+            assertThat(typeInfo.get("created")
                     .get("format"), is("DATE"));
-            assertThat((String) typeInfo.get("expiration")
+            assertThat(typeInfo.get("expiration")
                     .get("format"), is("DATE"));
-            assertThat((String) typeInfo.get("id")
+            assertThat(typeInfo.get("id")
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("title")
+            assertThat(typeInfo.get("title")
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("metadata-content-type")
+            assertThat(typeInfo.get("metadata-content-type")
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("metadata-content-type-version")
+            assertThat(typeInfo.get("metadata-content-type-version")
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("metadata-target-namespace")
+            assertThat(typeInfo.get("metadata-target-namespace")
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("resource-uri")
+            assertThat(typeInfo.get("resource-uri")
                     .get("format"), is("STRING"));
-            assertThat((Boolean) typeInfo.get("resource-uri")
+            assertThat(typeInfo.get("resource-uri")
                     .get("indexed"), is(true));
             // since resource-size is not indexed, it should be filtered out
-            assertThat((Boolean) typeInfo.get("resource-size")
+            assertThat(typeInfo.get("resource-size")
                     .get("indexed"), is(false));
-            assertThat((String) typeInfo.get("metadata")
+            assertThat(typeInfo.get("metadata")
                     .get("format"), is("XML"));
-            assertThat((String) typeInfo.get("location")
+            assertThat(typeInfo.get("location")
                     .get("format"), is("GEOMETRY"));
         }
     }
 
-    private CatalogFramework createFramework() {
-        final long COUNT = 2;
+    private CatalogFramework createPagingFramework(long count, int pagesExpected) {
+        CatalogFramework framework = mock(CatalogFramework.class);
+        List<Result> results = new ArrayList<>();
+
+        for (int i = 0; i < count * pagesExpected; i++) {
+            Result result = generateResult(i, count);
+            results.add(result);
+        }
+
+        QueryResponse firstResponse = new QueryResponseImpl(mock(QueryRequest.class),
+                new ArrayList<>(),
+                count);
+        for (int i = 0; i < count; i++) {
+            firstResponse.getResults()
```

Comment:
That is an elegant solution. Perhaps it should be moved to a location so that other paging improvements could use it in their tests...

I will adapt the logic in this class a bit to use some of it, but will try to keep it under the number of lines for my "dirty" implementation.